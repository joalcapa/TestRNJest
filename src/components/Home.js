import React, {Component} from 'react';
import { StyleSheet, View, TextInput, Button } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadTeams } from '../duck/actions';

type Props = {};
class Home extends Component<Props> {

    constructor(props) {
        super(props);
        console.log("store: ", this.props);
        //console.log("Store: ", this.props.getState());
        this.state = {
            data: "test"
        }
    }
/*
    componentDidMount() {
        Users.all().then((data) => {
         console.warn("chack api data", data);
         });
    }
*/
    onButtonPreset() {
       // console.log("Duck: ", Duck);
        this.props.loadTeams();
        console.log("Teams: ", this.props.teams);
        console.log("Go change", this.state.data);
    }

    render() {
        return (
            <View style={styles.container}>
                <Button
                    onPress={this.onButtonPreset}
                    title="Learn More"
                    color="#841584"
                    accessibilityLabel="Learn more about this purple button"
                />
            </View>
        );
    }
}

function mapStateToProps(state) {
    console.log("Se modifico el state: ", state);
    return {
        teams: state.teamsReducer.teams
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({ loadTeams }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    },
    welcome: {
        fontSize: 20,
        textAlign: 'center',
        margin: 10,
    },
    instructions: {
        textAlign: 'center',
        color: '#333333',
        marginBottom: 5,
    },
});