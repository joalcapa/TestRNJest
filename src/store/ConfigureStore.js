import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducers from '../duck/reducers';

export default function ConfigureStore () {
    const store = createStore(
        reducers,
        applyMiddleware(thunk)
    );

    return store;
};