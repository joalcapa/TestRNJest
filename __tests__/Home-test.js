import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/components/Home';
import { Provider } from 'react-redux';
import ConfigureStore from '../src/store/ConfigureStore';

let store = ConfigureStore();
const expect = global.expect;

jest.mock('Button', () => {
    const RealComponent = require.requireActual('Button')
    const React = require('React')
    class Button extends React.Component {
        render () {
            //this.props.getScrollableNode = function () {}
            return React.createElement('Button', this.props, this.props.children)
        }
    }
    Button.propTypes = RealComponent.propTypes
    return Button
})

describe("Component", () => {
    test('renders component Home', () => {

        const tree = renderer.create(
            <Provider store={store}>
                <Home />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});