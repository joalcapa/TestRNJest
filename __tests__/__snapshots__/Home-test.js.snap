// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders component Home 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Jose
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Alberto
  </Text>
  <Button
    accessibilityLabel="Learn more about this purple button"
    color="#841584"
    onPress={[Function]}
    title="load teams"
  />
</View>
`;
